django/contrib/auth/views.py

```diff
@@ -334,6 +346,117 @@ def password_reset_complete(request,
     return TemplateResponse(request, template_name, context)
 
 
+# New class-based password reset views
+# - PasswordResetView sends the mail
+# - PasswordResetDoneView shows a success message for the above
+# - PasswordResetConfirmView checks the link the user clicked and
+#   prompts for a new password
+# - PasswordResetCompleteView shows a success message for the above
+
+class PasswordContextMixin(object):
+    extra_context = None
+
+    def get_context_data(self, **kwargs):
+        context = super(PasswordContextMixin, self).get_context_data(**kwargs)
+        context['title'] = self.title
+        if self.extra_context is not None:
+            context.update(self.extra_context)
+        return context
+
+
+class PasswordResetView(PasswordContextMixin, FormView):
+    template_name = 'registration/password_reset_form.html'
+    form_class = PasswordResetForm
+    success_url = reverse_lazy('password_reset_done')
+
```

Comment:
I see the reasoning for the line break but the arguments in the other views are organized similarly. Maybe one group arranged alphabetically would be simplest?